gpu/octree/src/cuda/radius_search.cu

```diff
@@ -254,20 +257,33 @@ namespace pcl
                             __shfl_sync(0xFFFFFFFF, query.z, active_lane)
                         );
 
-                        length = TestWarpKernel(beg, active_query, radius2, length, out, length_left);
+                        length = TestWarpKernel(beg, active_query, radius2, length, out, sqr_dist, length_left);
                     }
                     else
                     {                            
                         length = min(length, length_left);                        
                         Warp::copy(batch.indices + beg, batch.indices + beg + length, out);
+
+                        if (batch.compute_all_distances)
```

Comment:
You also need to fill `sqr_dist` with nans in case compute all distances is set to false. aka mimic the same behavior existing in the cpu counterpart